### DISCUSSION TITLE
What is the best way to connect an existing record by id?

### DISCUSSION THREAD

**Discussion by nkeil** [prisma/prisma#18880] [Q&A] [answered]
When writing my queries, I usually use a syntax like this, where I just specify the id of an existing record to associate with my new record:
```ts
// User has many Posts
const newPost = await prisma.post.create({
  data: {
    userId: userId,
  },
});
```
But I sometimes see syntax written this way, using a call to connect:
```ts
const newPost = await prisma.post.create({
  data: {
    user: { connect: { id: userId } },
  },
});
```
Is there a preferred way of doing this? Are there cases where you'd want to use one over the other?

**Comment by nurul3101** [comment, marked as answer]
Hello @nkeil 👋 

You are correct, you could directly set the foreign key and use connect as well. Both should work.

However, it is better to use connect to link a foreign key because prisma would throw an error in case it cannot find the connecting record, so connect would provide an extra layer of validation.

**Reply by nkeil** [reply]
Makes sense, thanks for clearing that up!
